f/setGroupID/f_setGroupIDs.sqf

```diff
@@ -2,426 +2,35 @@
-["GrpSyn_HSAM1","Syndikat HSAM1 -"],
-["GrpSyn_ST1","Syndikat ST1 -"],
-["GrpSyn_DT1","Syndikat DT1 -"],
-["GrpSyn_ENG1","Syndikat ENG1 -"],
-
-["GrpSyn_IFV1","Syndikat IFV1 -"],
-["GrpSyn_IFV2","Syndikat IFV2 -"],
-["GrpSyn_TNK1","Syndikat TNK1 -"],
-
-["GrpSyn_TH1","Syndikat TH1 -"],
-["GrpSyn_TH2","Syndikat TH2 -"],
-["GrpSyn_TH3","Syndikat TH3 -"],
-["GrpSyn_TH4","Syndikat TH4 -"],
-["GrpSyn_TH5","Syndikat TH5 -"],
-["GrpSyn_TH6","Syndikat TH6 -"],
-["GrpSyn_TH7","Syndikat TH7 -"],
-["GrpSyn_TH8","Syndikat TH8 -"],
-["GrpSyn_AH1","Syndikat AH1 -"],
-
-// ====================================================================================
-
-// GROUP IDs: Independent > FIA
-// Automatically assigns intelligible names to groups
-
-["GrpIFIA_CO","FIA I CO -"],
-["GrpIFIA_DC","FIA I DC -"],
-["GrpIFIA_COV","FIA I COV -"],
-
-["GrpIFIA_ASL","FIA I ASL -"],
-["GrpIFIA_A1","FIA I A1 -"],
-["GrpIFIA_A2","FIA I A2 -"],
-["GrpIFIA_AV","FIA I AV -"],
-
-["GrpIFIA_BSL","FIA I BSL -"],
-["GrpIFIA_B1","FIA I B1 -"],
-["GrpIFIA_B2","FIA I B2 -"],
-["GrpIFIA_BV","FIA I BV -"],
-
-["GrpIFIA_CSL","FIA I CSL -"],
-["GrpIFIA_C1","FIA I C1 -"],
-["GrpIFIA_C2","FIA I C2 -"],
-["GrpIFIA_CV","FIA I CV -"],
-
-["GrpIFIA_JSL","FIA I JSL -"],
-["GrpIFIA_J1","FIA I J1 -"],
-["GrpIFIA_J2","FIA I J2 -"],
-["GrpIFIA_JV","FIA I JV -"],
-
-["GrpIFIA_MMG1","FIA I MMG1 -"],
-["GrpIFIA_MMG2","FIA I MMG2 -"],
-["GrpIFIA_HMG1","FIA I HMG1 -"],
-["GrpIFIA_MAT1","FIA I MAT1 -"],
-["GrpIFIA_MAT2","FIA I MAT2 -"],
-["GrpIFIA_HAT1","FIA I HAT1 -"],
-["GrpIFIA_MTR1","FIA I MTR1 -"],
-["GrpIFIA_MSAM1","FIA I MSAM1 -"],
-["GrpIFIA_HSAM1","FIA I HSAM1 -"],
-["GrpIFIA_ST1","FIA I ST1 -"],
-["GrpIFIA_DT1","FIA I DT1 -"],
-["GrpIFIA_ENG1","FIA I ENG1 -"],
-
-["GrpIFIA_IFV1","FIA I TECH1 -"],
-["GrpIFIA_IFV2","FIA I TECH2 -"],
-["GrpIFIA_TNK1","FIA I TNK1 -"],
-
-["GrpIFIA_TH1","FIA I TH1 -"],
-["GrpIFIA_TH2","FIA I TH2 -"],
-["GrpIFIA_TH3","FIA I TH3 -"],
-["GrpIFIA_TH4","FIA I TH4 -"],
-["GrpIFIA_TH5","FIA I TH5 -"],
-["GrpIFIA_TH6","FIA I TH6 -"],
-["GrpIFIA_TH7","FIA I TH7 -"],
-["GrpIFIA_TH8","FIA I TH8 -"],
-["GrpIFIA_AH1","FIA I AH1 -"]
-
-// Always make sure there's no comma after the last entry!
-
-// ====================================================================================
-
-// END OF THE ARRAY CONTAING ALL GROUPS
-// Do not comment or delete this line!
-];
-
-// ====================================================================================
-
 // SET GROUP IDS
 // Execute setGroupID Function for all factions
-{_x call f_fnc_setGroupID} forEach _groups;
+// Note: The actual group IDs are defined in f/groupMarkers/fn_groupData.sqf
+{
+	_x params ["_grp", "", "", "", "_id"];
+	if (_id != "") then {
+		[_grp, _id] call f_fnc_setGroupID;
+	}
+} forEach f_var_groupData_all;
+
+// ====================================================================================
+
+// Warn if there are groups with playable units that are not present in f_var_groupData_all
```

Comment:
How does this check interact this virtual units such as Zeus and spectators?

Reply:
It didn't print a warning. I'll re-test it to understand exactly what its behaviour is.

Reply:
Ok, checked the wiki for [playableUnits](https://community.bistudio.com/wiki/playableUnits)
> Does not contain units of sideLogic. 
So this is fine.